### PR TITLE
add progress endpoint

### PR DESCRIPTION
**Pull request by baieric**

### CODE REVIEW

File: sigopt/interface.py [hunk 1]
```diff
@@ -65,6 +65,7 @@ def __init__(self, requestor):
       endpoints=[
         ApiEndpoint(None, Experiment, 'POST', 'create'),
         ApiEndpoint(None, object_or_paginated_objects(Experiment), 'GET', 'fetch'),
+        ApiEndpoint(None, object_or_paginated_objects(Experiment), 'GET', 'progress'),
```
Comment (pfhayes) [inline sigopt/interface.py]: this should probably be an `AggregatedObservation` (or whatever name we want to give this)

File: sigopt/objects.py [hunk 2]
```diff
@@ -284,3 +284,9 @@ class Token(ApiObject):
   client = Field(six.text_type)
   experiment = Field(six.text_type)
   user = Field(six.text_type)
+
+
+class Progress(ApiObject):
```
Comment (pfhayes) [inline sigopt/objects.py]: there's already a `Progress` object :(

Comment (baieric) [inline sigopt/objects.py]: oh oops. I have no idea what to name this...

**Comment by alexandraj777** [comment]
Probably don't want to merge until we've decided on a name?

**Comment by baieric** [comment]
I agree

File: sigopt/interface.py [hunk 3]
```diff
@@ -59,6 +60,14 @@ def __init__(self, requestor):
       ],
     )
 
+    aggregatedObservations = ApiResource(
```
Comment (pfhayes) [inline sigopt/interface.py]: snake_case please

File: sigopt/objects.py [hunk 4]
```diff
@@ -284,3 +284,9 @@ class Token(ApiObject):
   client = Field(six.text_type)
   experiment = Field(six.text_type)
   user = Field(six.text_type)
+
+
+class AggregatedObservation(ApiObject):
+  assignments = Field(Assignments)
+  value = Field(float)
+  value_stddev = Field(float)
```
Comment (pfhayes) [inline sigopt/objects.py]: `failed` goes here too right?

Comment (pfhayes) [inline sigopt/objects.py]: i see... could a `failed` observation ever get returned by this endpoint? what if the user has only reported failed observations?

Comment (baieric) [inline sigopt/objects.py]: No, if all the observations are failed, none of them are selected as a "best" observation, so the endpoint returns an empty list

**Pull request merged**